akka-cluster/src/main/scala/akka/cluster/ClusterReadView.scala

```diff
@@ -85,9 +98,19 @@ private[akka] class ClusterReadView(cluster: Cluster) extends Closeable {
   def state: CurrentClusterState = _state
 
   def self: Member = {
+    _cachedSelf match {
+      case OptionVal.None ⇒
+        // lazy initialization here, later updated from elsewhere
+        _cachedSelf = OptionVal.Some(selfFromStateOrPlaceholder)
+        _cachedSelf.get
```

Comment:
might be racy, e.g. other thread set `_cachedSelf = None` inbetween, keep the `selfFromStateOrPlaceholder` result in a val

Reply:
hm, perhaps not since it's only updated by one actor, but wouldn't hurt to avoid

Reply:
Only place it is set to None is initially in constructor. So worst case here is that the first access and the first event about self comes concurrently, both leads to the latest member instance from state is set, so no problem?